eniius/mcstas.py

```diff
@@ -494,13 +523,81 @@ def NXtransformations(self, name):
     def NXcomponent(self, name, order=0):
         # Returns a NXcomponent corresponding to a McStas component.
         comp = self.components[self.indices[name]]
-        mcpars = {p:getattr(comp, p) for p in comp.parameter_names}
+        # pull together component values (or instrument parameter names) for *defined* parameters
+        # -- if a parameter is 'None' at this point, the Component default should be used
+        mcpars = {p: mcstasscript_parameter_name_or_value(getattr(comp, p)) for p in comp.parameter_names if getattr(comp, p) is not None}
         return McStasComp2NX(comp, order, self.NXtransformations(name), **mcpars).nxobj
 
     def NXinstrument(self):
         nxinst = NXinstrument()
+        nxinst['mcstas'] = mcstasscript_instrument_to_nx(self.instrument)
         for order, comp in enumerate(self.components):
             nxinst[comp.name] = self.NXcomponent(comp.name, order)
         return nxinst
 
 
+def _float_int_or_str(s: str):
+    # Parse a string to an integer if possible, or a float if numeric but not integer, or return the string
+    try:
+        v = float(s)
+        return int(v) if v.is_integer() else v
+    except ValueError:
+        return s
+        
+
+def mcstasscript_parameter_name_or_value(parameter):
+    from mcstasscript.helper.mcstas_objects import DeclareVariable
+    from libpyvinyl.Parameters.Parameter import Parameter
```

Comment:
I think global imports are to-be-avoided in general, since they would get picked up by anyone that later uses `from eniius.mcstas import *`. 
The runtime import overhead difference between global and local scoping is _probably_ small. So I would suggest keeping these function-scoped.